### PR TITLE
[bugfix] Fix bug when patchlevel in environment modules version is not an integer

### PR DESCRIPTION
**Pull request by jonathanfrawley**
A system I am working on uses version numbers such as "3.2.1+26-ad12ae47" for the module environment. This breaks reframe, but since reframe only looks at the major and minor numbers, it is not required to parse out the bugfix version at all.

### CODE REVIEW

File: reframe/core/modules.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def __init__(self):
 
         version = version_match.group(1)
         try:
-            ver_major, ver_minor, *_ = [int(v) for v in version.split('.')]
+            ver_major, ver_minor = [int(v) for v in version.split('.')[:2]]
```
Comment (vkarak) [inline reframe/core/modules.py:436]: I'rather do:

```suggestion
            ver_major, ver_minor, *_ = [int(v) for v in version.split('.', maxsplit=2)]
```

**Review by vkarak — CHANGES_REQUESTED** [review]
Hi @jonathanfrawley, thanks for the PR and the fix. We will also need a unit test to reproduce the bug. I can provide one.

**Comment by vkarak** [comment]
> Hi @jonathanfrawley, thanks for the PR and the fix. We will also need a unit test to reproduce the bug. I can provide one.

Ignore my comment about the unit tests. I just realised it was in the way we initialize the modules backend, so it should be fine if you just address my other comment.

**Review by vkarak — APPROVED** [review]
`maxsplit=2` won't work, so lgtm as it is now.

**Comment by vkarak** [comment]
ok to test

**Review by victorusu — APPROVED** [review]
`lgtm`

**Comment by vkarak** [comment]
@jenkins-cscs retry tsa

**Pull request merged**